shelley/chain-and-ledger/cddl-spec/shelley.cddl

```diff
@@ -4,6 +4,7 @@ block =
   [ header
   , transaction_bodies         : [* transaction_body]
   , transaction_witness_sets   : [* transaction_witness_set]
+  , transaction_metadata_set   : { * uint => transaction_metadata }
```

Comment:
I thought we were just going to have a generic `DataHash -> Data` witness map?

Reply:
The transaction body only needs a hash of the whole thing, and then there is no need to have hashes as keys.